core/src/main/scala/com/mesosphere/usi/core/matching/PortResource.scala

```diff
@@ -0,0 +1,223 @@
+package com.mesosphere.usi.core.matching
+
+import com.mesosphere.usi.core.ResourceUtil
+import com.mesosphere.usi.core.matching.PortResource._
+import com.mesosphere.usi.core.models.{ResourceMatchResult, ResourceRequirement, ResourceType}
+import com.mesosphere.usi.core.protos.ProtoBuilders
+import org.apache.mesos.v1.{Protos}
+
+import scala.annotation.tailrec
+import scala.collection.JavaConverters._
+import scala.util.Random
+
+case class PortResource(ports: Seq[Int], random: Random = Random) extends ResourceRequirement {
+  override def description: String = s"$resourceType:[${ports.mkString(",")}]"
+
+  override def resourceType: ResourceType = ResourceType.PORTS
+
+  @tailrec private def findResourceToMatch(
```

Comment:
Could you add docs. I have a hard time understand what this method does.

Reply:
Renamed and added docs...